### ISSUE TITLE
Wrong app version loaded in case of duplicate apps installations in multiple `apps_paths` entries

### ISSUE THREAD

**Issue by rubikonx9** [nextcloud/server#33794]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Since a while ago, I have multiple `apps_paths` entries configured:

>   array (
>    0 =>
>    array (
>      'path' => '/usr/share/webapps/nextcloud/apps',
>      'url' => '/apps',
>      'writable' => false,
>    ),
>    1 =>
>    array (
>      'path' => '/var/lib/nextcloud/apps',
>      'url' => '/wapps',
>      'writable' => true,
>    ),
>  ),

I had some apps installed in the 1st directory, then I created the 2nd directory and made the 1st one not writeable.
After installing new apps updates, the updates were installed in the 2nd directory, as it's the only one that is writeable.
The old app version (in the 1st directory) wasn't deleted, again, probably because of it not being writeable.
This resulted in two distinct copies of the app, each in different version.

Web UI / apps as well as `occ app:list` reports the new version, which is installed in the 2nd directory; however, when navigated to the Web UI of that app, nextcloud serves the old version, from the 1st directory. This is inconsistent.

### Steps to reproduce

I suspect that this is how it happened, I didn't really reproduce it more than once.

1. Install Nextcloud.
2. Install some app from the store.
3. Modify the configuration entry of `apps_paths` by marking the existing path as not writeable, and by creating a second dir which is writeable.
4. Update the app form the store.

### Expected behavior

The fact of having two versions of the app with the same name installed in parallel is, I think, expected, due to the former directory not being writeable anymore.

However, what I find problematic, is the inconsistency - the fact that a different version of the app is being loaded by the UI, and different one is being reported by Apps administration page and `occ`.

Perhaps it would be possible for `occ` to detect such situation and report a warning accordingly?
Generally, I found out about this problem nearly accidentally (when analyzing why an app I have stopped working) - it would be much better if Nextcloud admin panel or `occ` had told me about it.

### Installation method

Other Community project

### Operating system

Other

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MariaDB

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "htaccess.RewriteBase": "\/",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "24.0.4.1",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "maintenance": false,
        "theme": "",
        "loglevel": 0,
        "logfile": "\/var\/log\/nextcloud\/nextcloud.log",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "default_phone_region": "PL",
        "apps_paths": [
            {
                "path": "\/usr\/share\/webapps\/nextcloud\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/lib\/nextcloud\/apps",
                "url": "\/wapps",
                "writable": true
            }
        ],
        "app_install_overwrite": [
            "previewgenerator",
            "files_texteditor",
            "weather",
            "documentserver_community"
        ]
    }
}
```

### List of activated Apps

```shell
Enabled:
  - accessibility: 1.10.0
  - activity: 2.16.0
  - apporder: 0.15.0
  - bruteforcesettings: 2.4.0
  - calendar: 3.5.0
  - circles: 24.0.1
  - cloud_federation_api: 1.7.0
  - comments: 1.14.0
  - contacts: 4.2.0
  - contactsinteraction: 1.5.0
  - cookbook: 0.9.14
  - cospend: 1.4.8
  - dashboard: 7.4.0
  - dav: 1.22.0
  - documentserver_community: 0.1.12
  - drawio: 1.0.3
  - extract: 1.3.5
  - federatedfilesharing: 1.14.0
  - files: 1.19.0
  - files_markdown: 2.3.6
  - files_pdfviewer: 2.5.0
  - files_rightclick: 1.3.0
  - files_sharing: 1.16.2
  - files_texteditor: 2.14.0
  - files_trashbin: 1.14.0
  - files_versions: 1.17.0
  - files_videoplayer: 1.13.0
  - impersonate: 1.11.0
  - logreader: 2.9.0
  - lookup_server_connector: 1.12.0
  - maps: 0.1.10
  - metadata: 0.16.0
  - nextcloud_announcements: 1.13.0
  - notifications: 2.12.0
  - oauth2: 1.12.0
  - onlyoffice: 7.5.4
  - password_policy: 1.14.0
  - passwords: 2022.9.20-build4540
  - photos: 1.6.0
  - previewgenerator: 5.0.0
  - privacy: 1.8.0
  - provisioning_api: 1.14.0
  - recommendations: 1.3.0
  - riotchat: 0.13.5
  - serverinfo: 1.14.0
  - settings: 1.6.0
  - sharebymail: 1.14.0
  - support: 1.7.0
  - survey_client: 1.12.0
  - systemtags: 1.14.0
  - tasks: 0.14.4
  - text: 3.5.1
  - theming: 1.15.0
  - twofactor_backupcodes: 1.13.0
  - updatenotification: 1.14.0
  - user_status: 1.4.0
  - video_converter: 1.0.5
  - viewer: 1.8.0
  - weather: 1.7.6
  - weather_status: 1.4.0
  - workflowengine: 2.6.0
Disabled:
  - admin_audit
  - encryption
  - federation: 1.9.0
  - files_external
  - firstrunwizard: 2.8.0
  - user_ldap
```

### Nextcloud Signing status

```shell
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- INVALID_HASH
		- .htaccess
- passwords
	- EXTRA_FILE
		- css/CreatePassword.css
		- css/SettingsSection.css
		- css/vendors~HelpSection.css
		- css/StarChaser.css
		- css/ViewRevision.css
		- css/3bafc2f184a539e0dede926f31e6fbce.png
		- css/SetupWizard.css
		- css/Form.css
		- css/BackupSection.css
		- css/ConnectClient.css
		- css/HelpSection.css
		- css/AppsSection.css
		- css/023c9ea941ebacf354086bdf6faefc1d.svg
		- css/f028c722a78e240e77af95c62a178b57.gif
		- js/Static/SettingsSection.c1e301f5bd363b209562.js
		- js/Static/ImportManager.c1e301f5bd363b209562.js
		- js/Static/StarChaser.c1e301f5bd363b209562.js
		- js/Static/AppsSection.c1e301f5bd363b209562.js
		- js/Static/ExportManager.c1e301f5bd363b209562.js
		- js/Static/HelpSection.c1e301f5bd363b209562.js
		- js/Static/SetupWizard.c1e301f5bd363b209562.js
		- js/Static/vendors~ConnectClient.c1e301f5bd363b209562.js
		- js/Static/vendors~HelpSection.c1e301f5bd363b209562.js
		- js/Static/xlsx.c1e301f5bd363b209562.js
		- js/Static/vendors~xlsx.c1e301f5bd363b209562.js
		- js/Static/ViewRevision.c1e301f5bd363b209562.js
		- js/Static/Form.c1e301f5bd363b209562.js
		- js/Static/vendors~ExportManager.c1e301f5bd363b209562.js
		- js/Static/CreatePassword.c1e301f5bd363b209562.js
		- js/Static/vendors~HelpSection~marked~simplemde.c1e301f5bd363b209562.js
		- js/Static/vendors~simplemde.c1e301f5bd363b209562.js
		- js/Static/BackupSection.c1e301f5bd363b209562.js
		- js/Static/simplemde.c1e301f5bd363b209562.js
		- js/Static/ConnectClient.c1e301f5bd363b209562.js
		- js/Static/CsvHero.c1e301f5bd363b209562.js
		- appinfo/database.xml
		- lib/Exception/Migration/UpgradeUnsupportedException.php
		- lib/Exception/SecurityCheck/BreachedPasswordsFileAccessException.php
		- lib/Middleware/LegacyMiddleware.php
		- lib/Controller/Api/Legacy/LegacyVersionApiController.php
		- lib/Controller/Api/Legacy/LegacyCategoryApiController.php
		- lib/Controller/Api/Legacy/LegacyPasswordApiController.php
		- lib/Notification/LegacyApiNotification.php
		- lib/Helper/AppSettings/LegacyApiSettingsHelper.php
		- lib/Helper/Upgrade/UpgradeCheckHelper.php
		- img/browser/brave.png

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [.htaccess] => Array
                        (
                            [expected] => d9cab0f077161bf043d6d85f5bcacbdfaf82913968b7706df9cf7e6f98b11af363f3843b455175ae775567916822fec8f6ab95000ba064c71e05c86a02196d95
                            [current] => 691b3144e7851acf68ead8bb26373a3191ac9d476dc48e80123c0a87536480fd393596b590504012151e0860904c1bee9cb275c10c08d0b2fc1570727256a3ec
                        )

                )

        )

    [passwords] => Array
        (
            [EXTRA_FILE] => Array
                (
                    [css/CreatePassword.css] => Array
                        (
                            [expected] => 
                            [current] => 93ffa3fb92f2795bca312d72d0b6c3f6b6c9df38530534c9e1c8b9ddf28b3a59669a0413c34680bf09d98e5affaefab87bfbf5d09e8fa31274f5039d059b08b5
                        )

                    [css/SettingsSection.css] => Array
                        (
                            [expected] => 
                            [current] => 1967bf5efd049c190e870305488875aad8dad51ae8afdb3524d5cc36d1ba48397162741c4f9049cd4816c6597753bd4620e58cdcb15feab2ea668394de82206b
                        )

                    [css/vendors~HelpSection.css] => Array
                        (
                            [expected] => 
                            [current] => fff7cc2cee52ec552028c0a8dae94d6f30faa4a8cd57adf29fd22c03abdb6027b4f36a6bf3b8cf520fb6fe17bb4a79dea2f1be3677cee9b3ec5ae5558900f82d
                        )

                    [css/StarChaser.css] => Array
                        (
                            [expected] => 
                            [current] => 11e76a1b94d93aa3f96c11fd822d5c6c76044f873a82410def3fc8b99f77ef3851900fdb25b7ffd34fba8cf7d3cac1288c556b2a6704100f351ed706f1b71d48
                        )

                    [css/ViewRevision.css] => Array
                        (
                            [expected] => 
                            [current] => eb514a28b8cd66474bccbebd2d891e0ebe8044118888df2632554d22c4200c4c39037f1f46e443302bce8360f066d646c94e699c8a464ac2c1cf282aeeb3c44f
                        )

                    [css/3bafc2f184a539e0dede926f31e6fbce.png] => Array
                        (
                            [expected] => 
                            [current] => 239b0dcee7131399d66339d3e668e513d783f5f7270c77b917cf966b448d7729c4168178ad3ee88b30c590a5c3eab2582deacd668a7f624638654ff743550530
                        )

                    [css/SetupWizard.css] => Array
                        (
                            [expected] => 
                            [current] => 71d88c18ab8d81ca8e381779814100171528774f8aace13690626b94944d97e81f3d099d24a04b71bea2b1dc0031f4259aef6996787ea67dd70f56dfa0db3430
                        )

                    [css/Form.css] => Array
                        (
                            [expected] => 
                            [current] => 9c2e10c102a2e75a244cd3ceedf767e8ed0b125b01862d5b1bc95eb94989c3540e5e864b9537e7937e979b49d4c8583c95d721ee01579baa0fc8e06b1f551e93
                        )

                    [css/BackupSection.css] => Array
                        (
                            [expected] => 
                            [current] => 0e8be21824440f1672f9dc996a3861b271bd7a98acd71d7381634dc9d2bb9d602dc05699ce4bfd0e566b64db09e1b1a1200c3872c99fac1c76c7aa24c73bd6d5
                        )

                    [css/ConnectClient.css] => Array
                        (
                            [expected] => 
                            [current] => 24b70b9006ede5d2b358ba568e341723b77bacd68d4faaadca4a7bb9d826a906c9a69e5374709b5cf6c020a3e8b7d4d62f50501ba0e32dc62ed3012afaf24f0c
                        )

                    [css/HelpSection.css] => Array
                        (
                            [expected] => 
                            [current] => 559447e38e6bdcfe6f01c99fe76baec2b169457e84917de45ee2745afb0151790fe8bf6c0789b97b6f75a5ab72a9323297cb31a635621cc883e158dce7624e63
                        )

                    [css/AppsSection.css] => Array
                        (
                            [expected] => 
                            [current] => a6cde41a17e2f067232fb7255283ef9b75dbcdbf560faf82552200702738fab6bac452cd628f25a4a5f88c8edb7e41bd61787740bfd101ade2dcb12134cd7848
                        )

                    [css/023c9ea941ebacf354086bdf6faefc1d.svg] => Array
                        (
                            [expected] => 
                            [current] => e7aae36123a183e41ebe27f3bba76977a653e31c983abe48e2a4a234241c451395c9fd5d2220ef9035ba75d25c28f7e360a01ad0d4f8ecb1446557d944f5fa0f
                        )

                    [css/f028c722a78e240e77af95c62a178b57.gif] => Array
                        (
                            [expected] => 
                            [current] => 54084a4bef162b10b76a29e2505ffb30402656e093bc18d9d4745f1fed562edaebfa3630661e3cd60cbe5e128216b25123a1c563c79431b1700ce90fccdddaba
                        )

                    [js/Static/SettingsSection.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => 1b60a7cd4e695eb8ca9d362bd96ea1e4ff4c565283749bd3ab9f6aa26c359c6861a79ea8fcfbfc5665fb92391fafce6d95336762a02e92588da940a0d344ee0b
                        )

                    [js/Static/ImportManager.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => 36f9588f7f50abfe128ad23c109fd41b8bfcfc47ff60b3270d298167b24493c1844d89818add732cdad3c21b5b3e60ae3de37f652cc2d08dbeaaa294c36da1bf
                        )

                    [js/Static/StarChaser.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => b01aa92d49729162b33fbfe59cd523c29f95de06c1bf4489678798fccc5d950ccbf2be353f359b9a97dd90bb260abe584e2a3e4265a665050473aa9b51848d71
                        )

                    [js/Static/AppsSection.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => 5950a4d7acc1999790598ba6767fd36dd6b4f09311704fce17705858340ad8634fbe1a3c09a19beae92f4e2c9eff7c27c9f5466b31a179bd788919a350977702
                        )

                    [js/Static/ExportManager.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => b06af4fbd4ae423a597a9ededa95a511eaf4636ab6c54e745c6976d2d9a8e0eee45eb9804d6884982fd57362aea07287b6b0411ff5ab1c7f6472fe4b9fdafb5e
                        )

                    [js/Static/HelpSection.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => 74f19e9edaed9cb7a3d969eb9cf6f9235b545dbf4ab40e64b49128406a77c6070158fa7e46435dad26781be89b53e97a4e448f3065d1d496cb932826eafebf18
                        )

                    [js/Static/SetupWizard.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => 54fe28513aac904c499937c32389c3f9870be1d149563843ad65e923724092f20226f5088c929e3157d2492ccd9b59ba0248ef867374e20a3da3afbf9bb2d7da
                        )

                    [js/Static/vendors~ConnectClient.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => 6d3080e6d0e0fb6f7304e4cb3e6db41dd9d86ff3c3ec721a16d0ebf5d6f356d5395cfba80ddaadaca59d3e6cdf3204dc4bb2141d28254b452ea319ba95b4ff86
                        )

                    [js/Static/vendors~HelpSection.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => 73b4c343c9442bc6667da56cbcc2bba3000a74e96c0e1d07ac585a3b0ff4b18af33964142057462b0429c502c2a1650e2272f285968ca414032f2dcd2b38f941
                        )

                    [js/Static/xlsx.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => 23a1a562dc10369575cdb5d8d267b79be087a3b8f95e89b4887cc1c7d5d44b3009767c4b0a939d096821bd95f2d4941e904242c92c8af74a29db1c1a103fabfe
                        )

                    [js/Static/vendors~xlsx.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => 44b675a6b938aac704352681cb98987d2dbd5fdffc4b7d6cd95591a64199d83af61cfaae752745627983c8a664a74c8ceee81db097e3582d35ab33b4aa11f38c
                        )

                    [js/Static/ViewRevision.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => 6a8c25ced48f99f7f21790953413b3a3b029511cfcee0d9c784f7117fc4fcc77177473a27d02cb08c69cd1c836ebd457d574fea701590f42d954a634d9b53ae9
                        )

                    [js/Static/Form.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => 34cf311c29d18047e2c989c88658b3277a5cf9903ecbc961962cfb4ce46ef58211b8ac4294dc1060a42e92f10eef9db7b23e184f03e7efe4e211667ffc31aed7
                        )

                    [js/Static/vendors~ExportManager.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => e0d89efe5c22e4bae88f698089e3e0f5e3cd283b9ac467c734f4c29652a36ea40db7a74d701fd991aa4e2c33856b67516490c915ab98e2982efb9be1f50f70d6
                        )

                    [js/Static/CreatePassword.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => fbcaaf1456f2a5d2519a27dd46a2f0f72680261cb37edd124568d78ef294e3b2e906d2cbb611198e5438601af5998cbc598b495eba85b85128591161c3819f74
                        )

                    [js/Static/vendors~HelpSection~marked~simplemde.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => a617cdd50b087b67bb47b3b728932bc1f67ae1755d21f910343f54674fe56fd15ef9bd11da99ea2e39b70153770184f2601a3faed32aa30f12f85d3d92da236e
                        )

                    [js/Static/vendors~simplemde.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => ddac0b786030f464337e28109ca16691047e0d04b98481d12acaf4b965d650cedb1c52589c46a6eea57963d74ccbaa407176c035df2098a8aa1d011bdfeed6aa
                        )

                    [js/Static/BackupSection.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => 3e9fe9abc81d4e13e3ee7bc178a798e8737e9243e8930417d1c08727fae535767ab5765cba31e48df16f833325cdde7e2877607cf097ca563db280cd8e1910cc
                        )

                    [js/Static/simplemde.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => cc9d0fbb02d9350913767aa2c9497dc83cd0806dec944852dd244d26a4290bbdcbe89285f621a5a7b73cb6b086d4ec0e9a0b949512f4fbe9a1ffd87515f8029e
                        )

                    [js/Static/ConnectClient.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => 4d93f83c33c3b232c0d7532bdbcabf222d9165a05ab30abb4f9c9a1adbdffbf2b051538f2853b663d0dc941feb7574ab64674d7ad98a5f9ac924c244bd02107a
                        )

                    [js/Static/CsvHero.c1e301f5bd363b209562.js] => Array
                        (
                            [expected] => 
                            [current] => 6e4b49b75e210f84190cffca684276d48a65bc9c457d6309bc4d5ad825aaf17bd8c3b6935afa6f580843850d66956f067c3009b621e768594e4cd001f0b9dbc7
                        )

                    [appinfo/database.xml] => Array
                        (
                            [expected] => 
                            [current] => 64bff5ef863d9a34bf63652521c79773fd76c7b204094c3bac4ef23737263333590252143505bb321a2513cc84490a579c0ea738f6856aee716cb8d07c13112b
                        )

                    [lib/Exception/Migration/UpgradeUnsupportedException.php] => Array
                        (
                            [expected] => 
                            [current] => 276e5b380f4586b138470ff6d5f6c2086be45b98db5e6346682f7cafa364a794d20315466b68da97100df37e66a2b4710fd19a52651acc2b6ddbc2b11a6863fc
                        )

                    [lib/Exception/SecurityCheck/BreachedPasswordsFileAccessException.php] => Array
                        (
                            [expected] => 
                            [current] => dd0bcc91b12cbc4bd21c8026264aeebf0096183433da4212208d40bc052da9d977bdc17045b5e1a6b1ccb46aa3d2bc50f49d20a11660906b8ade0b2403d37b2a
                        )

                    [lib/Middleware/LegacyMiddleware.php] => Array
                        (
                            [expected] => 
                            [current] => eebe2458c30f73e0c584001ffa1048d7de3f2c696607032ef9f548383662c098c7eb52d9381092ef7e772e96707bb99aec9534dddf040affd0bd0ba1c66842c4
                        )

                    [lib/Controller/Api/Legacy/LegacyVersionApiController.php] => Array
                        (
                            [expected] => 
                            [current] => fb87cbcefa19a4161031d43048fed6c88e0f873ec2823eec0278850307bc7890b76674364497d0fbdaba7d0589ea0358be9a2ace6b94f19f7b4cc4cf75f3a5f8
                        )

                    [lib/Controller/Api/Legacy/LegacyCategoryApiController.php] => Array
                        (
                            [expected] => 
                            [current] => a4ecbbc43b9bf39d3e3ef75fe3a982e43ef2763a54163062ff42688ec1742da0487cc8f4673f93545702747bffc35dfcd9d1aa533ee2507b87318c84c9aa753c
                        )

                    [lib/Controller/Api/Legacy/LegacyPasswordApiController.php] => Array
                        (
                            [expected] => 
                            [current] => 4164686d4407c022f2593db50156e9335dcd3344d5ad83bcc98afd32b28026413bbf10c4d169b31a4276570bb3c2cf0b0f3ca0e9961559f8f0e63ef7356df9ae
                        )

                    [lib/Notification/LegacyApiNotification.php] => Array
                        (
                            [expected] => 
                            [current] => 09637f324ca0a20fe163ec5d6df34faa605eff11db77cce37d47170f258d76352b2da163d5329db14ac3664624b9be911f8f8ce5dd5e3425281fb365c33c8a5f
                        )

                    [lib/Helper/AppSettings/LegacyApiSettingsHelper.php] => Array
                        (
                            [expected] => 
                            [current] => 92908da827c0da55b3c12f2db76729b2d1d9a71ebda21bee9cd113b27ecfa25481cb0805f7748a448bc4066aa9f60ed1f4767205d1d021ff8602f0fe0d9e1a44
                        )

                    [lib/Helper/Upgrade/UpgradeCheckHelper.php] => Array
                        (
                            [expected] => 
                            [current] => a43ad2dbaacd74f86117c5aaa58a2c8e738a4dab7a06628d95aadba985e86c2738881759e2f3e3853c4f447225f55d2cf284413e46e6f433404251c3ff193802
                        )

                    [img/browser/brave.png] => Array
                        (
                            [expected] => 
                            [current] => a53c579013e39afdbe2f4b8b88136a925b29baadb4c6fbcf54474425ee3d65e65577a91ab8f614ec7298de51e2a55d53706dcdabed5bfa13eca8c734f2981e3b
                        )

                )

        )

)
```

### Additional info

I could see a relevant issue being mentioned here: https://github.com/nextcloud/contacts/issues/2867, there the root cause of the problem was also having multiple copies of an app.

**Comment by szaimen** [comment]
Hi, please update to 24.0.9 or better 25.0.3 and report back if it fixes the issue. Thank you!

My goal is to add a label like e.g. 25-feedback to this ticket of an up-to-date major Nextcloud version where the bug could be reproduced. However this is not going to work without your help. So thanks for all your effort!

If you don't manage to reproduce the issue in time and the issue gets closed but you can reproduce the issue afterwards, feel free to create a new bug report with up-to-date information by following this link: https://github.com/nextcloud/server/issues/new?assignees=&labels=bug%2C0.+Needs+triage&template=BUG_REPORT.yml&title=%5BBug%5D%3A+

**Comment by rubikonx9** [comment]
Hello,

Unfortunately, I don't consider the issue solved.
My current Nextcloud version: `25.0.3.2`.

I have made a test as follows:
I restored an older version (`2023.01.23`) of `Passwords` app into directory `/usr/share/webapps/nextcloud/apps/`, which is not writeable by `nextcloud` user.

Then I run `occ upgrade` (I also tried with `occ app:update --all`, same results).

The app was updated to newer version (`2023.02.20`), and stored in `/var/lib/nextcloud/apps/`, which is the second apps dir, which is writeable by `nextcloud` user.

`occ app:list` and Nextcloud GUI report the version installed to be `2023.02.20` - which is as expected.

However, when navigating to the app itself in Nextcloud GUI, it still attempts to load the older version, and fails to do so.
I verified this by checking the JS Console, and it's clear that it attempts to load older JS files (which is obvious looking at the files' names).

I couldn't see any warnings about duplicated app or inconsistent versions being present anywhere - neither in GUI or in `occ` commands.

Let me know when I can try again, or if you need any more info.

**Comment by joshtrichards** [comment]
Duplicate of #37043 

Let's consolidate tracking for this matter into #37043 since they're effectively the same and will likely be addressed in the same way.
